src/transformers/modeling_longformer.py

```diff
@@ -707,3 +707,141 @@ def forward(
             outputs = (masked_lm_loss,) + outputs
 
         return outputs  # (masked_lm_loss), prediction_scores, (hidden_states), (attentions)
+
+
+@add_start_docstrings(
+    """Longformer Model with a span classification head on top for extractive question-answering tasks like SQuAD (a linear layers on top of
+    the hidden-states output to compute `span start logits` and `span end logits`). """,
+    LONGFORMER_START_DOCSTRING,
+)
+class LongformerForQuestionAnswering(BertPreTrainedModel):
+    config_class = LongformerConfig
+    pretrained_model_archive_map = LONGFORMER_PRETRAINED_MODEL_ARCHIVE_MAP
+    base_model_prefix = "longformer"
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+
+        self.longformer = LongformerModel(config)
+        self.qa_outputs = nn.Linear(config.hidden_size, config.num_labels)
+
+        self.init_weights()
+
+    def _get_question_end_index(self, input_ids):
+        sep_token_indices = (input_ids == self.config.sep_token_id).nonzero()
+        assert sep_token_indices.ndim == 2
```

Comment:
Also would be nice if you could add some assert error messages here. 

1. The first assert we can prob remove - it just asserts that the `input_ids` has two dimensions.
The second one is actually very nice.
2. Something along the lines "There should be exactly three separator tokens in every sample for questions answering" 
3. IMO same as the first one. Maybe just add "input_ids should have two dimensions"

@ibeltagy - hope my analysis for the second assert is correct here

Reply:
@patrickvonplaten 
Yes, we can remove the first assert, `sep_token_indices.ndim` is always 2, even for single example.

But the third assert should come before the second, because when `input_ids` dim is not 2 then second assert fails.